Nos/Views/AuthorCard.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  AuthorCard.swift
+//  Nos
+//
+//  Created by Martin Dutra on 3/4/23.
+//
+
+import SwiftUI
+
+/// This view displays the information we have for an author suitable for being used in a list or grid.
+struct AuthorCard: View {
+
+    @ObservedObject var author: Author
+
+    @Environment(\.managedObjectContext) private var viewContext
+
+    var style = CardStyle.compact
+
+    var didTapGesture: (() -> Void)?
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            HStack(alignment: .center) {
+                Button {
+                    didTapGesture?()
+                } label: {
+                    HStack(alignment: .center) {
+                        AvatarView(imageUrl: author.profilePhotoURL, size: 24)
+                        Text(author.safeName)
+                            .lineLimit(1)
+                            .font(.subheadline)
+                            .foregroundColor(Color.primaryTxt)
+                            .multilineTextAlignment(.leading)
+                            .frame(maxWidth: .infinity, alignment: .leading)
+                        if author.muted {
```

Comment:
If the author is muted I think it makes sense to just return EmptyView() for this whole thing.

Reply:
Ok it's fine to leave for later then. Our mute code in general needs a rework I think.

Reply:
Sorry, probably my comment wasn't proactive enough. I mean, I'm fine with doing this here and in other places, but I  need to check that the final result is what we want.

The list of authors already filters muted authors, so this piece of code wouldn't be executed there. But as is a generic card, it could be used in other places in the future and I think it should be up to the view that uses this card to decide if it wants to show muted authors (with the muted label) or not.

Another example, this is similar of not equal to what FollowCard does. The FollowsView uses it to show all authors I follow. Should we show muted authors I follow? Right now we are showing them (unless muting unfollows the author), even if we wouldn't want to show muted authors, the change should be done in the FollowsView query.

Where I'm going to is, I think the code is fine as it is. There could be places where we would want to show muted authors, and in the places where we don't want to show them, the filter is done one level above.